testng-runner-api/src/main/java/org/testng/internal/TestResult.java

```diff
@@ -219,6 +219,9 @@ public void setThrowable(Throwable throwable) {
   /** @return Returns the endMillis. */
   @Override
   public long getEndMillis() {
+    if (m_status == SKIP) {
```

Comment:
It's a workaround which fixes the issue but not a fix.
Instead, you need to check where `m_endMillis` is set and fix the business logic there.

Reply:
@juherr Thanks for pointing that out... Fixed it. Please check once again.